tests/Builder/ObjectBuilderTest.php

```diff
@@ -0,0 +1,253 @@
+        );
+
+        $builder->build(stdClass::class, '');
+    }
+
+    /**
+     * Tests that ensure validated does not throw when the object is valid.
+     *
+     * @return void
+     *
+     * @throws \LoyaltyCorp\RequestHandlers\Exceptions\RequestValidationException
+     */
+    public function testEnsureValidated(): void
+    {
+        $request = new TestRequest();
+
+        $validator = new ValidatorStub();
+        $builder = $this->getBuilder(null, $validator);
+
+        $builder->ensureValidated($request);
+
+        // No exception was thrown.
+        $this->addToAssertionCount(1);
+    }
+
+    /**
+     * Tests that ensure validated throws when the object is invalid in PreValidate.
+     *
+     * @return void
+     *
+     * @throws \LoyaltyCorp\RequestHandlers\Exceptions\RequestValidationException
+     */
+    public function testEnsureValidatedThrows(): void
+    {
+        $request = new TestRequest();
+
+        $validator = new ValidatorStub([
+            [], // PreValidate
+            [ // Primary Validate
+                new ConstraintViolation(
+                    'Message',
+                    'Message',
+                    [],
+                    '',
+                    '',
+                    ''
+                )
+            ]
+        ]);
+
+        $builder = $this->getBuilder(null, $validator);
+
+        $this->expectException(RequestValidationException::class);
+
+        $builder->ensureValidated($request);
+    }
+
+    /**
+     * Tests that ensure validated throws when the object is invalid in PreValidate.
+     *
+     * @return void
+     *
+     * @throws \LoyaltyCorp\RequestHandlers\Exceptions\RequestValidationException
+     */
+    public function testEnsureValidatedThrowsPreValidate(): void
+    {
+        $request = new TestRequest();
+
+        $validator = new ValidatorStub([[
+            new ConstraintViolation(
+                'Message',
+                'Message',
+                [],
+                '',
+                '',
+                ''
+            )
+        ]]);
+
+        $builder = $this->getBuilder(null, $validator);
+
+        $this->expectException(RequestValidationException::class);
+
+        $builder->ensureValidated($request);
+    }
+
+    /**
+     * Gets the builder under test.
+     *
+     * @param \Symfony\Component\Serializer\SerializerInterface|null $serializer
+     * @param \Symfony\Component\Validator\Validator\ValidatorInterface|null $validator
+     *
+     * @return \LoyaltyCorp\RequestHandlers\Builder\ObjectBuilder
+     */
+    private function getBuilder(
+        ?SerializerInterface $serializer = null,
+        ?ValidatorInterface $validator = null
+    ): ObjectBuilder {
+        return new ObjectBuilder(
+            $serializer ?? new SerializerStub(),
+            $validator ?? new ValidatorStub()
+        );
+    }
+
+//    /**
```

Comment:
Needs a comment on why this is commented out. Ideally mark as a skipped test with a link to the ticket to fux.